tensorboard/webapp/feature_flag/directives/feature_flag_directive_test.ts

```diff
@@ -0,0 +1,136 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  Component,
+  DebugElement,
+  Input,
+  OnChanges,
+  ViewChild,
+} from '@angular/core';
+import {TestBed, fakeAsync, tick} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+import {Store} from '@ngrx/store';
+import {MockStore} from '@ngrx/store/testing';
+
+import {provideMockTbStore} from '../../testing/utils';
+import {FEATURE_FLAGS_HEADER_NAME} from '../http/const';
+import {getFeatureFlagsToSendToServer} from '../store/feature_flag_selectors';
+import {State as FeatureFlagState} from '../store/feature_flag_types';
+
+import {FeatureFlagDirective} from './feature_flag_directive';
+
+@Component({
+  selector: 'test',
+  template: `<p>{{ value }}</p>`,
+})
+export class TestComponent {
```

Comment:
This setup was necessary with the initial implementation (since ngOnChanges was used, but `detectChanges` wasn't invoking the lifecycle hook). I've simplified the setup to remove this.